src/catalog/column_catalog.cpp

```diff
@@ -165,28 +168,74 @@ bool ColumnCatalog::InsertColumn(oid_t table_oid,
   auto val6 = type::ValueFactory::GetBooleanValue(is_primary);
   auto val7 = type::ValueFactory::GetBooleanValue(is_not_null);
 
-  tuple->SetValue(ColumnId::TABLE_OID, val0, pool);
-  tuple->SetValue(ColumnId::COLUMN_NAME, val1, pool);
-  tuple->SetValue(ColumnId::COLUMN_ID, val2, pool);
-  tuple->SetValue(ColumnId::COLUMN_OFFSET, val3, pool);
-  tuple->SetValue(ColumnId::COLUMN_TYPE, val4, pool);
-  tuple->SetValue(ColumnId::IS_INLINED, val5, pool);
-  tuple->SetValue(ColumnId::IS_PRIMARY, val6, pool);
-  tuple->SetValue(ColumnId::IS_NOT_NULL, val7, pool);
-
+  auto constant_expr_0 = new expression::ConstantValueExpression(
+    val0);
+  auto constant_expr_1 = new expression::ConstantValueExpression(
+    val1);
+  auto constant_expr_2 = new expression::ConstantValueExpression(
+    val2);
+  auto constant_expr_3 = new expression::ConstantValueExpression(
+    val3);
+  auto constant_expr_4 = new expression::ConstantValueExpression(
+    val4);
+  auto constant_expr_5 = new expression::ConstantValueExpression(
+    val5);
+  auto constant_expr_6 = new expression::ConstantValueExpression(
+    val6);
+  auto constant_expr_7 = new expression::ConstantValueExpression(
+    val7);
+
```

Comment:
Smart pointers could be used instead. Memory management is a pain.

Reply:
Since the number of ConstantValueExpression objects to be allocated, is known at compile time - the seven different mallocs invocations can be replaced by a single invocation of malloc using an array of objects with dynamic initialization. 

eg: arr_const_expr = new expression::ConstantValueExpression[8]({val0}, {val2},...,{val7})